### PR TITLE
fix(filters): get inherit options from dashboard variables

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### CODE REVIEW

File: src/services/dashboards/dashboard-customize/modules/DashboardWidgetInputForm.vue [hunk 1]
```diff
@@ -212,13 +201,29 @@ export default defineComponent<Props>({
             if (Array.isArray(selectedItem)) return !!selectedItem.length;
             return selectedItem && !isEmpty(selectedItem);
         };
+        const getDashboardVariablesSchema = (propertySchema: JsonSchema['properties']) => {
+            const enabledVariables = Object.entries(dashboardDetailState.variablesSchema.properties)
+                .filter(([, d]) => {
+                    if (!d.use) return false;
+                    const variableType = d.selection_type === 'MULTI' ? 'array' : 'string';
+                    return propertySchema.type === variableType;
+                });
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/DashboardWidgetInputForm.vue:210]: description: Exclude options that do not match the dashboard variable type.

File: src/services/dashboards/dashboard-customize/modules/DashboardWidgetInputForm.vue [hunk 2]
```diff
@@ -212,13 +201,29 @@ export default defineComponent<Props>({
...
+            const _enum = enabledVariables.map(([key]) => key);
+            return {
+                type: 'string',
+                enum: _enum.length ? _enum : [null],
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/DashboardWidgetInputForm.vue:214]: description: ajv occurs error when enum is empty, so I set [null]. ([issue](https://github.com/ajv-validator/ajv/issues/824#issuecomment-723645085))

File: src/services/dashboards/dashboard-customize/stores/widget-form.ts [hunk 3]
```diff
@@ -57,10 +57,8 @@ export const useWidgetFormStore = defineStore('widget-form', () => {
             } else if (key.startsWith('filters.')) {
                 widgetFiltersMap[_propertyName] = [];
                 if (Array.isArray(val)) {
-                    val.forEach((v) => {
-                        if (v) widgetFiltersMap[_propertyName].push({ k: _propertyName, v, o: '' });
-                    });
-                } else if (val) widgetFiltersMap[_propertyName].push({ k: _propertyName, v: val, o: '' });
+                    widgetFiltersMap[_propertyName].push({ k: _propertyName, v: val, o: '' });
+                } else if (val) widgetFiltersMap[_propertyName].push({ k: _propertyName, v: val, o: '=' });
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/stores/widget-form.ts:61]: description: my mistake;;

```
// before
filter: [
  { k: 'provider', v: ['aws'], o: '' },
  { k: 'provider', v: ['azure'], o: '' },
  { k: 'provider', v: ['google_cloude'], o: '' },
]

// after
filter: [
  { k: 'provider', v: ['aws', 'azure', 'google_cloude'], o: '' },
]
```

**Review by WANZARGEN — APPROVED** [review]
LGTM!!

**Pull request merged**